crates/burn-jit/src/compute/kernel.rs

```diff
@@ -1,45 +1,93 @@
-use crate::kernel::{DynamicKernelSource, SourceTemplate, StaticKernelSource};
+#[cfg(feature = "extension")]
+use crate::template::SourceableKernel;
+use crate::{
+    gpu::{ComputeShader, WorkgroupSize},
+    kernel::{DynamicJitKernel, StaticJitKernel},
+    Compiler,
+};
 use alloc::sync::Arc;
 use core::marker::PhantomData;
 
-/// Kernel trait with the [source](SourceTemplate) that will be compiled and cached based on the
+/// Kernel for JIT backends
+///
+/// Notes: by default, only Jit variant exists,
+/// but users can add more kernels from source by activating the
+/// extension feature flag.
+pub enum Kernel {
+    /// A JIT GPU compute shader
+    Jit(Box<dyn JitKernel>),
```

Comment:
Could we simply have `Jit(Arc<gpu::ComputeShader>)` instead. I think we should remove as much as possible `Arc<dyn` and `Box<dyn` when not necessary. Maybe we need to create `GpuJitKernel`:

```rust
pub struct GpuJitKernel {
    pub id: String,
    pub shader: Arc<gpu::ComputeShapder>,
    pub info: ShaderInformation,
}
```
Therefore we would have a variant `JitGpu(JitGpuKernel)` and we can add more later instead of using dynamic traits.